### PR TITLE
Update to latest Twitch API

### PR DESCRIPTION
**Pull request by nanaya**
Now with more objects, abstract game name, ~~and shady regexp~~.

Also fixed featured stream video playback.

---

### CODE REVIEW

File: app/Models/LivestreamCollection.php [hunk 1]
```diff
@@ -36,17 +36,50 @@ public static function promote($id)
...
-    public function download()
+    public function downloadStreams()
+    {
+        return $this->download('streams?first=40&game_id=21465');
+    }
+
+    public function downloadUsers($userIds)
+    {
+        if (count($userIds) === 0) {
+        }
```
Comment (nekodex) [inline app/Models/LivestreamCollection.php]: Is there supposed to be something inside this if block? 🤔

File: app/Models/LivestreamCollection.php [hunk 2]
```diff
@@ -36,17 +36,50 @@ public static function promote($id)
...
+
+        $paramString = '';
+
+        foreach ($userIds as $id) {
+            $paramString .= "&id={$id}";
+        }
+
+        return $this->download("users?{$paramString}");
```
Comment (nekodex) [inline app/Models/LivestreamCollection.php]: Won't this result in `users?&id=...`?

File: app/Models/Twitch/Stream.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public function url()
+    {
+        return "https://twitch.tv/{$this->user['login']}";
+    }
+
+    public function preview($width, $height)
+    {
+        return str_replace(
```
Comment (nekodex) [inline app/Models/Twitch/Stream.php]: Maybe `strtr` instead to save nested `str_replace`?

File: app/Models/LivestreamCollection.php (hunk 2 above)
Comment (nanaya) [reply app/Models/LivestreamCollection.php]: Yes it will. No, it's not a problem.

File: app/Models/LivestreamCollection.php (hunk 1 above)
Comment (nanaya) [reply app/Models/LivestreamCollection.php]: That's a good question. And I can't remember what's this for ?_?

File: app/Models/LivestreamCollection.php (hunk 2 above)
Comment (nekodex) [reply app/Models/LivestreamCollection.php]: Why not just something like `return $this->download('users?id='.join($userIds, '&id='));`?

**Pull request merged**